\chapter {Literature Review}
\label{literature_review}
The folding process of amino acid interaction network is related to protein structure and protein folding process. Thus in this chapter we will discuss about some state of the art literature and algorithms on protein structure prediction and protein folding algorithms as well as about some previous work on amino acid interaction network, though there are few works on amino acid interaction.
\section {Protein Structure Prediction}
A grand challenge has been to develop a computer algorithm that can predict a protein’s 3D native structure from its amino acid sequence. On the one hand, knowledge of native structures is a starting point for understanding biological mechanisms and for discovering drugs that can inhibit or activate those proteins. On the other hand, we know 1000-fold more sequences than structures, and this gap is growing because of developments in high-throughput sequencing. So, there is considerable value in methods that could accurately predict structures from sequences.

Computer-based protein-structure prediction has been advanced by Moult and colleagues, in an event initiated in 1994 called CASP: Critical Assessment of protein Structure Prediction \cite{moult1,moult2}. Held every second summer, CASP is a community wide blind competition in which typically more than 100 different “target sequences” (of proteins whose structures are known but not yet publicly available) are made available to a community that numbers more than 150 research groups around the world. Each participating group applies some algorithmic scheme that aims to predict the 3D structures of these target proteins. After each CASP event, the true experimental structures are then revealed, group performances are evaluated, and community evaluations are published. 

Currently, all successful structure-prediction algorithms are based on assuming that similar sequences lead to similar structures. These methods draw heavily on the PDB, which now contains more than 80,000 structures. However, many of these structures are similar, and the PDB contains only $\sim$4000 structural families and 1200 folds \cite{murzin}.
CASP-wide progress over the past 18 years is summarized in Fig. \ref{fig:casp_evolution}. Prediction accuracies improved from CASP1 (1994) to CASP5 (2002) on the basis of several advances: (i) PDB expanded from ~1600 structures to 19,000 during that time. (ii) Better sequence search and alignment tools, such as Position-Specific Iterated Basic Local Alignment Search Tool (PSI-BLAST) \cite{altschul}, enabled the detection of more remote evolutionary relationships and more accurate sequence alignments. (iii) A strategy, called the “fragment assembly approach” \cite{jones,liam,simons,bonneau}, was developed that can often improve predictions when a similar sequence cannot be found in the PDB.
\begin{figure}[h!]
\centering
\subfloat [A. Historical CASP Performance]{\label{fig:casp_evolution} \includegraphics[width=0.5\textwidth]{casp_evolution}}
\subfloat [B. Performance in CASP9]{\label{fig:casp9_performance} \includegraphics[width=0.5\textwidth]{casp9_performance}}
\caption {Historical and present performance in CASP. Model quality is judged by using GDT-TS \cite{zemla}, which is approximately the percentage of residues that are located in the correct position. (A) Evolution of accuracy over the history of CASP, spanning 18 years. Each target is classified according to an approximate measure of difficulty that incorporates both the structural and sequence similarity to proteins of known structure \cite{kryshtafovych}. Each dot represents the best prediction (across all participants) for a given target. (B) Summary of prediction accuracy in CASP9 \cite{mariani}. We highlight the performance of two of the best automated server algorithms. Selected predictions are superimposed on the corresponding native structures to give a visual sense of the accuracy level that can be expected.}
\label {fig:casp_history}
\end{figure}
If the target protein’s sequence is related to a sequence that is already in the PDB, predicting its structure is usually easy (Fig. \ref{fig:casp_history}). In such cases, target protein structures are predicted by using “template-based modelling” (also called homology modelling or comparative modelling). But when there is no protein in the PDB with a sequence resembling the target’s, accurately predicting the structure of the target is much more difficult. These latter predictions are called “free modelling” (also called ab initio or de novo prediction). One of the most successful free-modelling techniques is fragment assembly, described below. 

In fragment assembly \cite{jones,liam,simons,bonneau}, a target protein sequence is de-constructed into small, overlapping fragments. A search of the PDB is performed to identify known structures of similar fragment sequences, which are then assembled into a full length prediction. The qualities of fragments and their assemblies are assessed by using some form of scoring function that aims to select more native-like protein structures from among the many possible combinations. Problems of folding physics described above share more commonality with free modeling than with template-based modelling.

Since CASP6, although overall progress has slowed (Fig. \ref{fig:casp_evolution}), there has been systematic, incremental progress \cite{kryshtafovych}. The best groups can now on average produce models that are better than the single best template from the PDB. Progress has been made toward successfully combining multiple templates into a single prediction. Substantial improvements have been observed for free-modelling targets shorter than 100 amino acids, although no single group yet consistently produces accurate models. Larger free-modeling targets remain challenging. Several recent algorithmic developments\textendash to predict residue\textendash residue contacts from sequence alone \cite{morcos,marks,jones2} and to more sensitively and accurately identify remote homologs \cite{remmert}\textendash promise to further improve prediction accuracy.
The performance of two of the best fully automated server predictors during CASP9 \cite{mariani} are shown in Fig. \ref{fig:casp9_performance}: HHPred, a pure template\textendash based modeling tool \cite{soding}, and ROSETTA, a hybrid tool that combines fragment assembly and template based modelling with all-atom refinement \cite{bradley}. For some fraction of CASP targets [$\sim$10\%, based on a cut-off of 85GlobalDistance Test–Total Score (GDT-TS) \cite{zemla2}, which is defined in Fig. \ref{fig:casp_history} , legend], the best predictions are now accurate enough to interpret biological mechanisms, to guide biochemical studies, or to initiate a drug discovery program (which requires structural errors of less than 2 to 3 \si{\angstrom}). However, it remains a challenge to predict the other 90\% of protein structures this accurately. In addition, it is also critical to improve physics-based technologies and to reduce our dependence on knowledge of existing structures, so that we can ultimately study protein motions, intrinsically disordered proteins, induced-fit binding of drugs, and membrane proteins and fold-able polymers, for which databases are too limited.
\section {Protein Folding}
Protein folding is a quintessential basic science. There has been no specific commercial target, yet the collateral pay-off's have been broad and deep. Specific technical advances are reviewed elsewhere \cite{dill}; below, we describe a few general outgrowths. 
\subsection{Growth of protein\textendash structure databases}
Today, more than 80,000 protein structures are known at atomic detail and publicly available through the PDB. New structures are being added at a rapid pace, supported by the National Institutes of Health (NIH)\textendash funded Protein Structure Initiative, which was developed in part to inform protein structure prediction.
\subsection{Advances in computing technology}
Understanding protein folding was a key motivation for IBM’s development of the Blue Gene supercomputer \cite{allen2001blue}, now also used to study the brain, materials, weather patterns, and quantum and nuclear physics. Protein folding has also driven key advances in distributed\textendash grid computing, such as in Folding@home, developed by Pande at Stanford, in which computer users all over the world donate their idle computer time to perform physical simulations of protein systems \cite{shirts2006screen}. Folding@home, which now has more than one million registered users and an average of 200,000 user\textendash donated CPUs available at any one time, provided some of the earliest simulations showing that MD simulations can accurately predict folding rates \cite{snow2002absolute}. The Anton computer from DE Shaw Research, custom designed to simulate biomolecules, gives several orders of magnitude better performance than conventional computers \cite{shaw2007anton}. Advances in computer technology have led to major advances in forcefields and to more reliable atomic\textendash level insights into biological mechanisms.
\subsection {Improvements in bio-molecular forcefields}
Computer processing power has advanced at the Moore’s law rate, doubling every $\sim$2 years. But equally important, forcefields have kept pace. Increased computer power leads to longer computed time scales, which puts more stringent demands on the accuracies of bio-molecular forcefields. In a pioneering paper in 1977, McCammon \textit{et al.} showed that the BPTI protein was stable in computer simulations during a computed time of 10 ps \cite{mccamm0n1977dynamics}. Today, small proteins are typically stable in explicit\textendash water simulations for 5 to 8 orders of magnitude longer\textendash microseconds to milliseconds of computed time \cite{shaw2010atomic}. Achieving such advances has required continuous improvements in forcefield accuracy.
\subsection {New sociological structures in the scientific enterprise}
Protein folding has driven innovations in how other field of science is done. CASP was among the first community-wide scientific competitions/collaborations, a paradigm for how grand\textendash challenge science can be advanced through an organized communal effort. Other such competitions have followed, including Critical Assessment of Prediction of Interactions (CAPRI) (predicting protein-protein docking) \cite{chen2003protein},SAMPL (predicting small-molecule solvation free energies, and ligand binding modes and affinities) \cite{guthrie2009blind}, and GPCR-Dock (predicting structures for G\textendash protein coupled receptors, a pharmaceutically important category of membrane proteins) \cite{michino2009community}, among many others. Protein folding has also pioneered “citizen science,” such as in Folding \cite{shirts2006screen} and Robetta@home and in a computer game called Foldit \cite{cooper2010predicting}, in which the public engages in protein folding on their home computers.
\begin{figure}[h]
\centering
\subfloat [Designed Proteins]{\label{fig:inhibitor} \includegraphics[width=0.5\textwidth]{inhibitor}}
\subfloat [Designed Foldamers]{\label{fig:molecular_paper} \includegraphics[width=0.5\textwidth]{molecular_paper}}
\caption {Designed proteins and foldamers. (A) A protein inhibitor that was designed by computer to bind to hem-agglutinin, an influenza protein. After design, the inhibitor was crystallized in a complex with hemagglutinin. The designed structure is in remarkably good agreement with experiment, particularly for the side chains involved in binding. (B) Peptoids are synthetic, fold-able, protein\textendash inspired polymers that have various applications. Shown here are peptoids that were designed as chains of alternating hydrophobic (gray) and either positively (blue) or negatively (red) charged side chains that spontaneously forms thin 2D structure called molecular paper.}
\label {fig:protein_foldamers}
\end{figure}
\subsection {New materials: Sequence-specific fold-able polymers}
The principles and algorithms developed for protein folding have led to non-biological, human-made proteins and to new types of polymeric materials. In particular, proteins have been designed that bind to and inhibit other proteins (fig. \ref{fig:inhibitor}) \cite{fleishman2011computational}, have new folds \cite{kuhlman2003design}, have new enzymatic activities \cite{siegel2010computational}, and act as potential new vaccines \cite{azoitei2011computation}. Also, a class of non-biological polymers has emerged, called “foldamers,” that are intended to mimic protein structures and functions \cite{gellman1998foldamers,horne2008foldamers,kirshenbaum1999designing,lee2005folding}. Foldamers already have broad ranging applications \cite{fowler2009structure,lee2008biomimetic,shah2008photoresponsive,nam2010free} as inhibitors of protein\textendash protein interactions, broad\textendash spectrum antibiotics, lung surfactant mimics, optical storage materials, a zinc-finger–like binder, an RNA protein binding disrupter for application in muscular dystrophy, gene transfection agents, and “molecular paper” (Fig. \ref{fig:molecular_paper}). Although such materials have potential applications in biomedicine and materials science, they also provide a way for us to test and deepen our understanding of protein folding.
