\chapter {Amino Acid Interaction Network}
\label{interaction_network}
Many systems, both natural and artificial, can be represented by networks, that is, by sites or vertices bound by links. The study of these networks is interdisciplinary because they appear in scientific fields like physics, biology, computer science or information technology.
These studies are lead with the aim to explain how elements interact with each other inside the network and what are the general laws which govern the observed network properties. From physics and computer science to biology and social sciences, researchers have found that a broad variety of systems can be represented as networks, and that there is much to be learned by studying these networks. Indeed, the studies of the Web \cite{broder2000graph}, of social networks \cite{wasserman1994social} or of metabolic networks \cite{jeong2000large} contribute to put in light common non-trivial properties of these networks which have a priori nothing in common. The ambition is to understand how the large networks are structured, how they evolve and what are the phenomena acting on their constitution and formation.
\section {Interaction Network General Models} 
In this chapter we present the three main models of interaction networks by describing their specific properties. We also define several measures that we use in the in order to study $SSE-IN$ (Secondary Structure Element Interaction Network, discussed in \ref{subsec:ssein}) empirically.
\subsection {Random Graph Model}
The random graph models are one of the oldest network models, introduced in \cite{solomonoff1951connectivity} and further studied in \cite{erdds1959random} and \cite{erd6s1960evolution}. These works identify two different classes of random graphs, called $G_{n,u}$ and $G_{n,p}$ and
defined by the following connection rules:
\begin{itemize}
\item $G_{n,u}$ regroups all graphs with $n$ vertices and $m$ edges. To generate a graph sampled uniformly at random from the set $G_{n,u}$, one has to put $m$ edges between vertex pairs chosen randomly from $n$ initial unconnected vertices.
\item $G_{n,p}$ is the set of all graphs consisting of $n$ vertices, where each vertex is connected to others with independent probability $p$. To generate a graph sampled randomly, one has to begin with $n$ initially unconnected vertices and join each pair by an edge with probability $p$.
\end{itemize}

In $G_{n,u}$ the number of edges is fixed whereas in $G_{n,p}$ the number of edges can fluctuate but its average is fixed. When $n$ tends to be large the two models are equivalent.
\textbf{Definition 1} The degree of a vertex $v, k_v$, is the number of edges incident to $v$. The mean degree, $z$, of a graph $G$ is defined as follows:
\[ z = \frac{1}{n} \sum_{v\in V} k_v = \frac{2m}{n} = p(n-1)\]
The degree distribution is one of the important characteristics of this kind of networks because it affects their properties and behaviour \cite{albert2000topology}. The random graph $G_{n,p}$ has a binomial degree distribution. The probability $p_k$ that a randomly chosen vertex is connected to exactly $k$ others is \cite{newman2001structure} :
\[ p_k = {n \choose k} p^k (1 - p)^{n-k} \]
when $n$ is tends to infinity, this becomes:
\[ p_k = \lim_{n \to \infty} \frac{n^k}{k!} {(\frac{p}{1-p})}^k (1-p)^n \approx \frac {z^k e^{-z}} {k!}\]

\begin{figure}[h]
\centering
\includegraphics[width=0.7\textwidth]{node_distribution-1}
\caption {Poisson distribution $p_k = \frac{z^k e^{-z}}{k!}$ with $z = 1, 2 $ and $4$.}
\label {fig:node_distribution-1}
\end{figure}
As we see in Figure \ref{fig:node_distribution-1}, Poisson distribution have different behaviour for different mean degree $z$. Each distribution has a clear peak close to $k = z$, followed by a tail that decays as $1 / k!$ which is considerably quicker than exponential.
\subsection {Small-World Networks}
This network model was introduced in \cite{watts1999small} as a model of social networks. It has been since adopted to treat phenomena in physics, computer science or social sciences. The model comes from the observation that many real-world networks have the following two properties:

\begin{itemize}
\item The small-world effect, meaning that most pairs of vertices are connected by a short path through the network. This phenomenon has two explanations. First, the concept of \enquote{shortcuts} through a network allows to join two distant vertices by a small number of edges \cite{watts1999small}.
Second, the concept of \enquote{hubs}, vertices whose connectivity is higher than others provide bridges between distant vertices because most vertices are linked to them.
\item High \enquote{clustering}, meaning that there is a high probability that two vertices are connected one to another if they share the same neighbour.
\end{itemize}
To determine if a network is a small-world, one can use the measures described below and compare them to the corresponding measures of a random graph.

\textbf{Definition 2} The characteristic path length \cite{watts1999small}, denoted $L$, of a graph $G$ is the median of the means of the shortest path lengths connecting each vertex $v$ to all other vertices. More precisely, let $d(v, u)$ be the length of the shortest path between two vertices $v$ and $u$ and let $\overline{d_v}$ be the average of $d(u,v)$ over all $u \in V$. Then the characteristics path length is the median of \{$\overline{d_v}$\}

This definition applies when the graph consists of single connected component. However, the $SSE-IN$ we consider in the next section may have several connected components. In this case, when we calculate the mean of the shortest path lengths $\overline{d(v)}$ we take into account only the vertices $u$ which are in the same connected component as $v$.

Since the mean and the median are practically identical for any reasonably symmetric distribution, the characteristic path length of a random graph is the mean value of the shortest path lengths between any two vertices. The characteristic path length of a random graph with mean degree $z$ is 
\[ L_{RG} = \frac{log n} {log z} \]
It increases only logarithmically with the size of the network and remains therefore small even for large systems.

\textbf{Definition 3} The local clustering coefficient \cite{watts1999small}, $C_v$, of a vertex $v$ with $k_v$ neighbours measures the density of the links in the neighbourhood of $v$.
\[ C_v = \frac{|E({\Gamma}_v)|}{{k_v \choose 2}} \]
where the numerator is the number of edges in the neighbourhood of $v$ and the denominator is the number of all possible edges in this neighbourhood. The clustering coefficient $C$ of a graph is the average of the local clustering coefficients of all vertices:
\[ C = \frac{1}{n} \sum_{v\in V} C_v\]
The clustering coefficient of a random graph with mean degree z is 
\[ C_{RG} = \frac{z}{n-1}\]
Watts and Strogatz \cite{watts1999small} define a network to be a small-world if it shows both of the following properties:
\begin{enumerate}
\item Small world effect: $L \approx L_{RG}$
\item High clustering: $C >> C_{RG}$
\end{enumerate}

\subsection {Scale-Free Networks}
The most important property of scale-free systems is their invariance to changes in scale. The term \enquote{scale-free} refers to a system defined by a functional form $f(x)$ that remains unchanged within a multiplicative factor under rescaling of the independent variable $x$. Indeed, this means power-law forms, since these are the only solutions to $f(an) = b f(n)$, where $n$ is the number of vertices \cite{newman2002structure}. The scale-invariance property means that any part of the scale-free network is stochastically similar to the whole network and parameters are assumed to be independent of the system size \cite{jeong2000large}.

If $n_k$ is the number of vertices having the degree $k$, we define $p_k$ as the fraction of vertices that have degree $k$ in the network:
\[ p_k = \frac{n_k}{n}\]
The degree distribution can be expressed via the cumulative degree function \cite{newman2002structure,erdds1959random}:
\[ P_k = \sum_{k' = k}^{\infty} p_{k'}\]
which is the probability for a node to have a degree greater or equal to $k$.

By plotting the cumulative degree function one can observe how its tail evolves, following a power law or an exponential distribution.

The power law distribution is defined as following \cite{newman2002structure}:
\[ P_k \approx \sum_{k'=k}^{\infty} k'^{\alpha} \approx k^{-(\alpha-1)}\]
and the exponential distribution is defined by the next formula:
\[ P_k \approx \sum_{k'=k}^{\infty} e^{-k' / \alpha} \approx e^{-k / \alpha}\]
Between this two distributions, there is a mixture of them where the distribution has a power law regime followed by a sharp cut-off, with an exponential decay of the tail, expressed by the next formula:
\[ P_k \approx \sum_{k'=k}^{\infty} k'^{-\alpha} e^{-k' / \alpha} \approx k^{\alpha-1} e^{-k / \alpha}\]

Like a power law distribution, it decreases polynomially, so that the number of vertices with weak degree is important while a reduced proportion of vertices having high degree exists. The last are called \enquote{hubs} that is sites with large connectivity through the network, as Figure \ref{fig:degree_distribution}.
\begin{figure}[h]
\centering
\includegraphics[width=0.6\textwidth]{degree_distribution}
\caption {Degree distribution described in \cite{amaral2000classes}. The red line follows a power law, as for scale-free networks. The green line corresponds to truncated scale-free networks. The black curve corresponds to single-scale networks.}
\label {fig:degree_distribution}
\end{figure}
The scale-free model depends mainly on the kind of degree distribution, thus a network is defined as a scale-free if:
\begin{itemize}
\item The degree distribution is a power law distribution $P(k) \approx k^{-\alpha}$ over a part of its range.
\item The distribution exposant satisfies $2 < \alpha \le 3$ \cite{goh2001universal}.
\end{itemize}
Amaral \textit{et al.} \cite{amaral2000classes} have studied networks whose cumulative degree distribution shape lets appear three kinds of networks. First, scale-free networks whose distribution decays as a power law with an exposant $\alpha$ satisfying bounds seen above. Second, as Figure \ref{fig:degree_distribution}, broad-scale or truncated scale-free networks whose the degree distribution has a power law regime followed by a sharp cut off. Third, single-scale networks whose degree distribution decays fast like an exponential.
\subsection {Topological Measures}
\label{subsec:topological_measures}
Here, we present some measures that we use to describe proteins' $SSE-IN$. Among them, there are simple ones, the most frequently used, but also more subtle, which allow a more precise discrimination between interaction networks.
\paragraph {Diameter and mean distance} The distance in a graph $G = (V,E)$ between two vertices $u,v \in V$, denoted by $d(u,v)$, is the length of the shortest path connecting $u$ and $v$ \cite{3161009}. If there is no path between $u$ and $v$, we suppose that $d(u,v)$ is undefined. A graph diameter, $D$, is the longest shortest path between any two vertices of a graph \cite{3161009}:
\[ D = max \{d(u,v) | u, v \in V \}\]
The mean distance is defined as the average distance between each couple of vertices:
\[ \overline{d_G} = \frac{2}{n(n-1)} \sum_{u,v\in V} d(u,v)\]
\paragraph{Density} The density, denoted $\delta$, is defined as the ratio between the number of edges in a graph and the maximum number of edges which it could have:
\[ \delta (G)=\frac{2m}{n(n-1)}\approx \frac{2m}{n^2}\]
The density of a graph is a number between 0 and 1. When the density is close to one, the graph is called dense, when it is close to $0$, the graph is called sparse \cite{coleman1983estimation}.
\paragraph{Clustering coefficients} Watts and Strogatz proposed a measure of clustering \cite{watts1999small} and defined it as a measure of local vertices density, thus for each node $v$, the local clustering around its neighbourhood is defined in the following way:
\[ C_v = \frac{1}{2}k_v(k_v-1)\]
The clustering coefficient is a ratio between the number of edges and the maximum number of possible edges in the vertices neighbourhood. If we extend the previous definition to the entire graph, the clustering is given by the expression:
\[ C_{local} = \frac{1}{n}\sum_{v\in V}\frac{N_{connected}}{C_v}\]
where, $N_{connected}$ is number of connected neighbour pairs.
\paragraph{} The last definition is mainly local because for each node, it involves only its neighbourhood. The global clustering was studied by Newman \textit{et al.} \cite{newman2001structure} and can be measured by the following formula:
\[ C_{global} = \frac{3 \times N_{triangle}}{N_{triplet}}\]
where, $N_{triangle} $ is number of triangle in the graph and $N_{triplet}$ is number of connected triplets of vertices.
\paragraph{} A triangle is formed by three vertices which are all connected and a triplet is constituted by three nodes and two edges. The global clustering coefficient $C_{global}$ is the mean probability that two vertices that are neighbours of the same other vertex will themselves be neighbours.
\section{Topological Description}
The behaviour or SSE-IN is studied in \cite{gaci2008proteins} published by Gaci \textit{et al.},which we are going to describe in this section. We want to observe how proteins from a same structural family provide similar SSE-IN according to their topological properties. To do that, we propose topological measures which we apply on a sample of proteins to put in evidence the existence of equivalence between structural similarity and topological homogeneity in the resulting SSE-IN.

The first step before studying the proteins SSE-IN is to select them according to their SSE arrangements. Thus, a protein belongs to a CATH topology level or a SCOP fold level iff all its domains are the same. We have worked with the CATH v3.1.0 and SCOP 1.7.1 files. We have computed the measures from the previous section for three families of each hierarchical classification, namely SCOP and CATH as in Table. \ref{tab:cath_family} and Table. \ref{tab:scop_family}. We have chosen these three families by classification, in particular because of their huge protein number. Thus, each family provides a broad sample guaranteed more general results and avoiding fluctuations. Moreover, these six families contain proteins of very different sizes, varying from several dozens to several thousands amino acids in SSE.
\begin{table}
\centering
\begin {tabular}{ | l | c | c |}
\hline
Name & Class & Proteins \\ \hline
RossmannFold & $\alpha$ $\beta$ & 2576 \\ \hline
TIM Barrel & $\alpha$ $\beta$ & 1051 \\ \hline
Lysozyme & Mainly $\alpha$ & 871 \\ \hline
\end {tabular}
\caption {CATH type studied protein family}
\label {tab:cath_family}
\end{table}

\begin{table}
\centering
\begin {tabular}{ | l | c | c |}
\hline
Name & Class & Proteins \\ \hline
Globin-like & All $\alpha$ & 733 \\ \hline
TIM $\beta$ / $\alpha$ -barrel & $\alpha$ / $\beta$ & 896 \\ \hline
Lysozyme-like & $\alpha$+$\beta$ & 819 \\ \hline
\end {tabular}
\caption {SCOP type studied protein family}
\label {tab:scop_family}
\end{table}

\subsection{Diameter and mean distance}
\begin{table}
\centering
\begin {tabular}{ | l | c |}
\hline
Protein Family Name & Diameter \\ \hline
RossmannFold & 18.84 \\ \hline
TIM Barrel & 19.83 \\ \hline
Lysozyme & 12.81 \\ \hline
\end {tabular}
\caption {Average diameter for CATH type studied protein family}
\label {tab:diameter_cath_family}
\end{table}

\begin{table}
\centering
\begin {tabular}{ | l | c |}
\hline
Protein Family Name & Diameter \\ \hline
Globin-like & 15.65 \\ \hline
TIM $\beta$ / $\alpha$ -barrel & 20.09 \\ \hline
Lysozyme-like & 12.85 \\ \hline
\end {tabular}
\caption {Average diameter for SCOP type studied protein family}
\label {tab:diameter_scop_family}
\end{table}
As we can see the average diameter for each $1$ of the studied family in Table. \ref{tab:diameter_cath_family} and \ref{tab:diameter_scop_family}, we can observe very close diameters between $TIM Barrel$ and $TIM \beta/\alpha-barrel$ and also between $Lysozyme$ and $Lysozyme-like$ families. This is explained by the fact that each pair of families contain almost the same proteins, in  other words, $Lysozyme$ topology in CATH is the equivalent of $Lysozyme-like$ fold level in SCOP.
\begin{figure}
\centering
\subfloat [Average Diameter of Rossman fold]{\label{fig:Rossmanfold_distribution} \includegraphics[width=0.5\textwidth]{Rossmanfold_distribution}}
\subfloat [Average Diameter of $\beta$/$\alpha$-barrel]{\label{fig:globin_distribution} \includegraphics[width=0.5\textwidth]{globin_distribution}}
\caption {Average Diameter Distribution}
\label {fig:diameter_distribution}
\end{figure}
If we closely observe the Figure. \ref{fig:diameter_distribution}, where distribution of diameter of the two studied families are depicted, the distribution follows roughly a Poison law. These result confirm that the mean diameter is suitable property to discriminate families between them.

The diameter being an upper bound of distances in interaction networks, we expect that the mean distance $z$ will be lower than $D$. Table. \ref{tab:average_mean_distance} confirms this. Again, we observe very close values between the equivalent SCOP and CATH families for the reasons discussed above. But we can also see that different families have values which allow discrimination between them based on this parameter. It is interesting to note that the ratio $D / z$ is about $2.5$ for all the families. The last property is a characterization of all proteins' $SSE-IN$.
\begin{table}
\centering
\begin {tabular}{ | l | c |}
\hline
Protein Family Name & $\overline{d_G}$ \\ \hline
RossmannFold & 7.26 \\ \hline
TIM Barrel & 7.79 \\ \hline
Lysozyme & 4.99 \\ \hline
\hline
Globin-like & 6.64 \\ \hline
TIM $\beta$ / $\alpha$ -barrel & 7.86 \\ \hline
Lysozyme-like & 5.03 \\ \hline
\end {tabular}
\caption {Average of mean distances for each family}
\label {tab:average_mean_distance}
\end{table}
\subsection{Density and mean degree}
As defined earlier, the density measures the ratio between the number of available edges and the number of all possible edges. Results presented in Table \ref{tab:average_density} show that the two families $TIM Barrel$ and $TIM \beta/\alpha -barrel$ have the minimum density. It has a consequence on their $SSE-IN$ topology. When the density is low, the network is less connected and consequently, the diameter and the average distance are higher. Comparing these results to Tables \ref{tab:diameter_scop_family} and \ref{tab:average_mean_distance} one can see the inversely proportional relation between density in one hand, and diameter and average distance on the other.
\begin{table}
\centering
\begin {tabular}{ | l | c |}
\hline
Protein Family Name & $\delta (G)$ \\ \hline
RossmannFold & 0.033 \\ \hline
TIM Barrel & 0.03 \\ \hline
Lysozyme & 0.038 \\ \hline
\hline
Globin-like & 0.034 \\ \hline
TIM $\beta$ / $\alpha$ -barrel & 0.029 \\ \hline
Lysozyme-like & 0.042 \\ \hline
\end {tabular}
\caption {Average of density for each family}
\label {tab:average_density}
\end{table}
\begin{table}
\centering
\begin {tabular}{ | l | c |}
\hline
Protein Family Name & $z$ \\ \hline
RossmannFold & 7.2 \\ \hline
TIM Barrel & 7.12 \\ \hline
Lysozyme & 6.82 \\ \hline
\hline
Globin-like & 7.69 \\ \hline
TIM $\beta$ / $\alpha$ -barrel & 7.15 \\ \hline
Lysozyme-like & 6.81 \\ \hline
\end {tabular}
\caption {Average of mean degree for each family}
\label {tab:average_mean_degree}
\end{table}
The mean degree, $z$, is presented in Table \ref{tab:average_mean_degree}. The observed values are close enough from one family to another. That is why the mean degree is not discriminating property, but rather a property characterizing all proteins' SSE-IN.

\subsection {Degree Distribution}
We compute the cumulative degree distribution for all proteins SSE-IN of studied families. A sample of our results is presented on Figure. \ref{fig:cumulative_degree_distribution}. We can remark that the curves follow a power law distribution and can be approximated by the following power-law function:
\[ p_k = 141.29k^{-\alpha} \] 
where $\alpha \approx 2.99$.

We observe the same results for all studied proteins. To explain this phenomenon, we have to rely on two facts. First, the mean degree of all proteins SSE-IN is nearly constant (Table. \ref{tab:average_mean_degree}). Second, the degree distribution, see Figure \ref{fig:degree_distribution1}, follows a Poisson distribution whose peak is reached for a degree near $z$. These two facts imply that for degree lower than the peak the cumulative degree distribution decreases slowly and after the peak its decrease is fast compared to an exponential one. Consequently, all proteins SSE-IN studied have a similar cumulative degree distribution which can be approximated by a unique power-law function.
\begin{figure}
\centering
\subfloat [1RXC cumulative distribution]{\label{fig:degree_distribution_1rx} \includegraphics[width=0.5\textwidth]{degree_distribution_1rx}}
\subfloat [1HV4 cumulative distribution]{\label{fig:degree_distribution_1hv4} \includegraphics[width=0.5\textwidth]{degree_distribution_1hv4}}
\caption {Cumulative degree distribution for a) 1RXC from Rossman fold and b) 1HV4 from TIM $\beta / \alpha$-barrel}
\label {fig:cumulative_degree_distribution}
\end{figure}
\begin{figure}
\centering
\subfloat [1RXC degree distribution]{\label{fig:degree_distribution_1rx1} \includegraphics[width=0.5\textwidth]{degree_distribution_1rx1}}
\subfloat [1HV4 degree distribution]{\label{fig:degree_distribution_1hv41} \includegraphics[width=0.5\textwidth]{degree_distribution_1hv41}}
\caption {Degree distribution for a) 1RXC from Rossman fold and b) 1HV4 from TIM $\beta / \alpha$-barrel}
\label {fig:degree_distribution1}
\end{figure}
\subsection {Clustering Coefficients}
The local clustering $C_{local}$ measures the fraction of pairs of a vertex's neighbours and the global clustering $C_{global}$ gives the probability that among three vertices at least two are connected. The results presented in Table \ref{tab:clustering_coefficient} show that the clustering coefficients are close for different families and cannot be correlated to density values. Consequently, the neighbour density remains independent of the previously studied properties.

\begin{table}
\centering
\begin {tabular}{ | l | c | c |}
\hline
Protein Family Name & $C_{local}$ & $C_{global}$ \\ \hline
RossmannFold & 0.63 & 0.56 \\ \hline
TIM Barrel & 0.64 & 0.57 \\ \hline
Lysozyme & 0.65 & 0.58 \\ \hline
\hline
Globin-like & 0.63 & 0.57 \\ \hline
TIM $\beta$ / $\alpha$ -barrel & 0.64 & 0.57 \\ \hline
Lysozyme-like & 0.66 & 0.58 \\ \hline
\end {tabular}
\caption {Clustering coefficients for each family}
\label {tab:clustering_coefficient}
\end{table}

\subsection{Consequences of the discussion}
In this part we introduce the notion of interaction network of amino acids of a protein (SSE-IN) and study some of the properties of these networks. We give different means to describe a protein structural family by characterizing their SSE-IN. Some of the properties, like diameter and density, allow discriminating two distinct families, while others, like mean degree and power law degree distribution, are general properties of all SSE-IN. Thus, proteins having similar structural properties and biological functions will also have similar SSE-IN properties. In this way our model allows us to draw a parallel between biology and graph theory.

\section {Amino Acid Interaction Network}
Though the term amino acid interaction network not so well known in the field of proteomics, it is now a good model to research in protein folding.
The 3D structure of a protein is determined by the coordinates of its atoms. This information is available in Protein Data Bank (PDB) \cite{berman2000protein}, which regroups all experimentally solved protein structures. Using the coordinates of two atoms, one can compute the distance between them. We define the distance between two amino acids as the distance between their $C\alpha$ atoms. Considering the $C\alpha$ atom as a “center” of the amino acid is an approximation, but it works well enough for our purposes. Let us denote by $N$ the number of amino acids in the protein. A contact map matrix is a $N \times N$ $0-1$ matrix, whose element $(i,j)$ is one if there is a contact between amino acids $i$ and $j$ and $zero$ otherwise. It provides useful information about the protein. For example, the secondary structure elements can be identified using this matrix. Indeed, $\alpha-$helices spread along the main diagonal, while $\beta-$sheets appear as bands parallel or perpendicular to the main diagonal \cite{ghosh2007dynamics}. There are different ways to define the contact between two amino acids. In \cite{gaci2009small}, the notion is based on spacial proximity, so that the contact map can consider non\textendash covalent interactions. Gaci \textit{et al.} says that two amino acids are in contact iff the distance between them is below a given threshold. A commonly used threshold is 7 \si{\angstrom}.
Consider a graph with $N$ vertices (each vertex corresponds to an amino acid) and the contact map matrix as incidence matrix. It is called contact map graph. The contact map graph is an abstract description of the protein structure taking into account only the interactions between the amino acids. Now let us consider the subgraph induced by the set of amino acids participating in $SSE$. We call this graph $SSE$ interaction network ($SSE-IN$) and this is the object we study in the present chapter. The reason of ignoring the amino acids not participating in $SSE$ is simple. Evolution tends to preserve the structural core of proteins composed from $SSE$. In the other hand, the loops (regions between $SSE$) are not so important to the structure and hence, are subject to more mutations. That is why homologous proteins tend to have relatively preserved structural cores and variable loop regions. Thus, the structure determining interactions are those between amino acids belonging to the same $SSE$ on local level and between different $SSE$s on global level. 
In \cite{muppirala2006simple} and \cite{brinda2005network} the authors rely on similar models of amino acid interaction networks to study some of their properties, in particular concerning the role played by certain nodes or comparing the graph to general interaction networks models. Thanks to this point of view the protein folding problem can be tackled by graph theory approaches.